core/src/main/java/com/microsoft/applicationinsights/internal/channel/samplingV2/SamplingScoreGeneratorV2.java

```diff
@@ -0,0 +1,61 @@
+package com.microsoft.applicationinsights.internal.channel.samplingV2;
+
+import com.microsoft.applicationinsights.agent.internal.common.StringUtils;
+import com.microsoft.applicationinsights.telemetry.Telemetry;
+
+import java.util.Random;
+
+/**
+ * Created by Dhaval Doshi Oct 2017
+ * This class generates the sample using the random number generator.
+ * It also contains the logic to preserve the correlated telemetry items.
+ */
+public class SamplingScoreGeneratorV2 {
+
+    private static Random random = new Random();
+
+    /**
+     * This method takes the telemetry and returns the hash of the operation id if it is present already
+     * or uses the random number generator to generate the sampling score.
+     * @param telemetry
+     * @return
+     */
+    public static double getSamplingScore(Telemetry telemetry) {
+
+        double samplingScore = 0.0;
+
+        if (!StringUtils.isNullOrEmpty(telemetry.getContext().getUser().getId())) {
+            samplingScore = (double) (getSamplingHashCode(telemetry.getContext().getUser().getId()) / Integer.MAX_VALUE);
+        }
+
+        else if (!StringUtils.isNullOrEmpty(telemetry.getContext().getOperation().getId())) {
+            samplingScore = (double) (getSamplingHashCode(telemetry.getContext().getOperation().getId()) / Integer.MAX_VALUE);
+        }
+
+        else {
+            long val = Math.abs(random.nextLong());
+            samplingScore =  ((double)Math.abs(val)/ Long.MAX_VALUE);
+        }
+
+        return samplingScore * 100;
+    }
+
+    private static int getSamplingHashCode(String input) {
```

Comment:
tests for this? Pls copy all the test cases from .NET. We found discrepancy with JavaScript on how numbers are processed. So having those test cases is important.

Reply:
Yes for sure. I will add for this.